### PR TITLE
adjust read timeout for skydns. default is too high. (changed 4 secs -> 2 secs)

### PR DESCRIPTION
**Pull request by ArtfulCoder**

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
Read https://github.com/kubernetes/kubernetes/issues/14394#issuecomment-143592443 for details on why this is being done.

**Comment by ArtfulCoder** [comment]
In addition to this change, we need to reduce the search domains that we have or configure skydns to not forward queries that end in "cluster.local". That will help mitigate the problem significantly.

**Comment by ArtfulCoder** [comment]
https://github.com/kubernetes/kubernetes/issues/14628

**Comment by ArtfulCoder** [comment]
150ms timeout might be too aggressive per this article:
https://technet.microsoft.com/en-us/library/ff807396(v=ws.10).aspx

**Comment by thockin** [comment]
Yeah, I think the REAL problem is the forwarding of queries which can never be resolved upstream.  Are we sure it's doing that?  The docs imply it should not be..

**Comment by ArtfulCoder** [comment]
per https://github.com/kubernetes/kubernetes/issues/14394#issuecomment-143872809, the timeout of skydns is effectively 4 seconds.
This change sets it back to 2 seconds.

@thockin PTAL, would like this in for 1.1

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 1]
```diff
@@ -58,6 +58,7 @@ spec:
         # command = "/skydns"
         - -machines=http://localhost:4001
         - -addr=0.0.0.0:53
+        - -rtimeout=1s
```
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: What happens when SkyDNS fixes their bug and this suddenly is "too short" ?

Comment (ArtfulCoder) [inline cluster/addons/dns/skydns-rc.yaml.in]: We will re-adjust / remove this flag if/when we pick the latest skydns release.

Also, hopefully, we will stop using skydns as a forwarder, making this flag redundant.

Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: Please file an issue, assign to yourself with the details.

**Comment by ArtfulCoder** [comment]
Merging this PR. We want this to be part of v1.1.
It will help mitigate DNS issues that users have been facing for a long time.

**Comment by thockin** [comment]
Please cherry pick to v1.1 and file that issue to fix this!!

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 1 above)
Comment (ArtfulCoder) [inline cluster/addons/dns/skydns-rc.yaml.in]: filed #14807

Comment (ArtfulCoder) [inline cluster/addons/dns/skydns-rc.yaml.in]: fixed

**Pull request merged**